### ISSUE TITLE
Fix flattening of POMs

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#369]
### Expected behavior

POMs need to be consistent and used to be flattened.

### Actual behavior

After #368 the flattening does not take place leading to broken POMs pointing to parent with `dev-SNAPSHOT` version ending up in maven central.

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. devon release (or mvn clean install)
2. observe resulting POMs

### Related/Dependent Issues

* #368 

### Comments/Hints:

Fix flattening

### Affected version:

3.0.0

**Comment by hohwille** [comment]
This is an interesting bug as actually the flattening takes place:
```
[INFO] Installing /Users/hohwille/projects/devon/workspaces/main/ide/eclipse-m2e/.flattened-pom.xml to /Users/hohwille/.m2/repository/com/devonfw/tools/ide/dev/devonfw-ide-eclipse-m2e/dev-SNAPSHOT/devonfw-ide-eclipse-m2e-dev-SNAPSHOT.pom
```
However, for this specific artifact the version is still `dev` and `-SNAPSHOT`. As I traced this down I found that `revision` and `changelist` are set to these values (`dev` and `-SNAPSHOT`) in the toplevel POM. I fix this to avoid confusion.
However, the specific artifact `devonfw-ide-eclipse-m2e` is not meant to be deployed to maven central and explicitly has this `dev-SNAPSHOT` version on purpose.
So I am not sure if the issue has already been fixed before or my tidy up will solve this.
Still I am not 100% sure why we deployed this to maven central with invalid parent:
https://repo1.maven.org/maven2/com/devonfw/tools/ide/devonfw-ide-scripts/3.3.0/devonfw-ide-scripts-3.3.0.pom

**Comment by hohwille** [comment]
The problem seems to be that the packaging is actually `pom` and `flatten-plugin` by default does not flatten if packaging is `pom` to prevent breaking BOMs or other POMs that should be deployed as is. So we need to explicitly set `flatten.mode` here.

**Comment by hohwille** [comment]
Gotcha! It was exactly this line that got lots with the update to `maven-parent`:
https://github.com/devonfw/ide/blob/07d846181a33cb2c62ab3c005329ec951bfb463f/pom.xml#L56

**Comment by hohwille** [comment]
Fixed with PR #387
